glommio/src/executor/stall.rs

```diff
@@ -0,0 +1,283 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+//
+#![allow(dead_code)]
+
+use nix::sys;
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread::JoinHandle,
+    time::{Duration, Instant},
+};
+
+#[cfg(feature = "stall-detection")]
+thread_local! {
+    static STALL_STRACE_COLLECTOR: (
+        crossbeam::channel::Sender<backtrace::BacktraceFrame>,
+        crossbeam::channel::Receiver<backtrace::BacktraceFrame>
```

Comment:
why do you need two channels? This flows in a single direction

Reply:
That's one channel; you're looking at the sender/receiver pair.